compilers/concrete-compiler/compiler/lib/Common/Keysets.cpp

```diff
@@ -417,5 +418,98 @@ KeysetCache::getKeyset(const Message<concreteprotocol::KeysetInfo> &keysetInfo,
   return std::move(keyset);
 }
 
+Message<concreteprotocol::KeysetInfo> generate_generic_keyset_info(
+    std::vector<concrete_optimizer::utils::PartitionDefinition> partitionDefs,
+    bool generateFks) {
+  auto output = Message<concreteprotocol::KeysetInfo>{};
+  rust::Vec<concrete_optimizer::utils::PartitionDefinition> rustPartitionDefs{};
+  for (auto def : partitionDefs) {
+    rustPartitionDefs.push_back(def);
+  }
+  auto parameters = concrete_optimizer::utils::generate_generic_keyset_info(
+      rustPartitionDefs, generateFks);
+
+  auto skLen = (int)parameters.secret_keys.size();
```

Comment:
could be outlined `CircuitKeys` to `Message<concreteprotocol::KeysetInfo>` I guess we should do that somewhere else in the code base ?

Reply:
After checking we use another struct `TFHE::TFHECircuitKeys` and the code here
https://github.com/zama-ai/concrete/blob/80ca8d3a395cd6f84313e3ef5ab396bfb88b7be8/compilers/concrete-compiler/compiler/lib/Support/ProgramInfoGeneration.cpp#L183

Not sure about the motivation of doing that, we could discuss later

Reply:
But that show variance is not set (and we maybe we should check on keygen that the variance is set...)